core/lib/spree/testing_support/dummy_app.rb

```diff
@@ -3,23 +3,17 @@
 ENV['RAILS_ENV'] = 'test'
 ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK'] = '1'
 
-require 'rails'
-require 'active_record/railtie'
-require 'action_controller/railtie'
-require 'action_mailer/railtie'
+require 'rails/all'
```

Comment:
ActiveStorage is available on Rails 5.2+, instead of using a complex `if` with checks on the versions `rails/all` already takes care of requiring everything you get the current Rails version. Further checks are then just based on `defined? ActiveStorage`.

Let me know if it needs a separate commit 👍